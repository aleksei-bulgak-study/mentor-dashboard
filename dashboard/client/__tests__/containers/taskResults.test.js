import React from 'react';
import { shallow } from 'enzyme';
import TaskResultsContainer from '../../src/containers/taskResults';

describe('TaskResultsContainer', () => {
  it('default snapshot', () => {
    // given
    const props = {
      task: { name: 'test', status: 'Checking' },
      students: ['student1', 'student2'],
      results: [
        {
          github: 'https://github.com/student1',
          tasks: {
            test: { mark: 100, pr: 'https://example.com/task' },
            test2: { mark: 30, pr: 'https://example.com/task2' },
          },
        },
      ],
    };
    // when
    const wrapper = shallow(<TaskResultsContainer {...props} />);
    // then
    expect(wrapper).toMatchSnapshot();
  });

  it('filterStudentResults', () => {
    // given
    const results = [
      {
        github: 'https://github.com/student1',
        tasks: {
          test: { mark: 100, pr: 'https://example.com/task' },
          test2: { mark: 30, pr: 'https://example.com/task2' },
        },
      },
    ];
    // when
    const instance = shallow(<TaskResultsContainer students={[]} results={[]} task={({})} />)
      .instance();
    // then
    expect(instance.filterStudentResults('student1', results)).toEqual({
      test: { mark: 100, pr: 'https://example.com/task' },
      test2: { mark: 30, pr: 'https://example.com/task2' },
    });
    expect(instance.filterStudentResults('student2', results)).toEqual({});
  });
});
