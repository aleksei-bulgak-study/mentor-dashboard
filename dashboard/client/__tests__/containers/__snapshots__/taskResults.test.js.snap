// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskResultsContainer default snapshot 1`] = `
<Fragment>
  <TaskResultContainer
    key="teststudent1"
    results={
      Object {
        "test": Object {
          "mark": 100,
          "pr": "https://example.com/task",
        },
        "test2": Object {
          "mark": 30,
          "pr": "https://example.com/task2",
        },
      }
    }
    student="student1"
    task={
      Object {
        "name": "test",
        "status": "Checking",
      }
    }
  />
  <TaskResultContainer
    key="teststudent2"
    results={Object {}}
    student="student2"
    task={
      Object {
        "name": "test",
        "status": "Checking",
      }
    }
  />
</Fragment>
`;
